rust/src/database/verkle/state.rs

```diff
@@ -0,0 +1,528 @@
+
+            for (expected, received) in code.iter().zip(code_buf[..retrieved_len].iter()) {
+                // SAFETY: `get_code` initializes `len` bytes of `code_buf`.
+                unsafe {
+                    assert_eq!(
+                        expected,
+                        &received.assume_init(),
+                        "code mismatch for case: {name}"
+                    );
+                }
+            }
+        }
+    }
+
+    #[rstest_reuse::apply(all_state_impls)]
+    fn setting_basic_data_fields_does_not_interfere_with_others(
+        #[case] state: Box<dyn CarmenState>,
+    ) {
+        let addr = Address::from_index_values(0, &[(0, 1)]);
+        let balance = crypto_bigint::U256::from_u32(42).to_be_bytes();
+        let nonce = 7u64.to_be_bytes();
+        let code = vec![0x01, 0x02, 0x03, 0x04, 0x05];
+
+        // Initially, all fields should be zero
+        let stored_balance = state.get_balance(&addr).unwrap();
+        assert_eq!(stored_balance, U256::default());
+        let stored_nonce = state.get_nonce(&addr).unwrap();
+        assert_eq!(stored_nonce, Nonce::default());
+        let stored_code_len = state.get_code_len(&addr).unwrap();
+        assert_eq!(stored_code_len, 0);
+
+        // Set balance
+        set_balance(&*state, addr, balance);
+        let stored_balance = state.get_balance(&addr).unwrap();
+        assert_eq!(stored_balance, balance);
+        let stored_nonce = state.get_nonce(&addr).unwrap();
+        assert_eq!(stored_nonce, Nonce::default());
+        let stored_code_len = state.get_code_len(&addr).unwrap();
+        assert_eq!(stored_code_len, 0);
+
+        // Set nonce
+        set_nonce(&*state, addr, nonce);
+        let stored_balance = state.get_balance(&addr).unwrap();
+        assert_eq!(stored_balance, balance);
+        let stored_nonce = state.get_nonce(&addr).unwrap();
+        assert_eq!(stored_nonce, nonce);
+        let stored_code_len = state.get_code_len(&addr).unwrap();
+        assert_eq!(stored_code_len, 0);
+
+        // Set code
+        set_code(&*state, addr, &code);
+        let stored_balance = state.get_balance(&addr).unwrap();
+        assert_eq!(stored_balance, balance);
+        let stored_nonce = state.get_nonce(&addr).unwrap();
+        assert_eq!(stored_nonce, nonce);
+        let stored_code_len = state.get_code_len(&addr).unwrap();
+        assert_eq!(stored_code_len, code.len() as u32);
+    }
+
+    #[rstest_reuse::apply(all_state_impls)]
+    fn can_store_and_retrieve_storage_values(#[case] state: Box<dyn CarmenState>) {
+        let set_storage = |state: &dyn CarmenState, addr: Address, key: Key, value: Value| {
+            state
+                .apply_block_update(
+                    0,
+                    Update {
+                        slots: &[SlotUpdate { addr, key, value }],
+                        ..Default::default()
+                    },
+                )
+                .unwrap();
+        };
+
```

Comment:
set_balance and set_nonce are freestanding functions, so let's keep it consistent.
```suggestion
    fn set_storage(state: &dyn CarmenState, addr: Address, key: Key, value: Value) {
        state
            .apply_block_update(
                0,
                Update {
                    slots: &[SlotUpdate { addr, key, value }],
                    ..Default::default()
                },
            )
            .unwrap();
    }

    #[rstest_reuse::apply(all_state_impls)]
    fn can_store_and_retrieve_storage_values(#[case] state: Box<dyn CarmenState>) {
```